src/app/services/error-handler.service.ts

```diff
@@ -0,0 +1,217 @@
+    private dialog: DialogService,
+    private translate: TranslateService,
+  ) {}
+
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  handleError(error: any): void {
+    if (this.isTypeOfWebsocketError(error)) {
+      const errors = this.parseWsError(error);
+      this.dialog.error(errors);
+      return;
+    }
+    if (this.isTypeOfJobError(error)) {
+      const errors = this.parseJobError(error);
+      this.dialog.error(errors);
+      return;
+    }
+    console.error('Error', error);
+  }
+
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  isTypeOfWebsocketError(obj: any): boolean {
+    return 'error' in obj
+    && 'extra' in obj
+    && 'reason' in obj
+    && 'trace' in obj;
+  }
+
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  isTypeOfJobError(obj: any): boolean {
+    return 'state' in obj
+    || 'error' in obj
+    || 'exception' in obj
+    || 'exc_info' in obj;
+  }
+
+  /**
+   * @param error
+   * @param formGroup
+   * @param fieldsMap Overrides backend field names with frontend field names.
+   * TODO: See if second `string` in fieldsMap can be typed to key of formGroup.
+   */
+  handleWsFormError(
+    error: WebsocketError | Job,
+    formGroup: UntypedFormGroup,
+    fieldsMap: Record<string, string> = {},
+  ): void {
+    if ('type' in error && error.type === ResponseErrorType.Validation && error.extra) {
+      this.handleValidationError(error, formGroup, fieldsMap);
+      return;
+    }
+
+    if ('exc_info' in error && error.exc_info.type === ResponseErrorType.Validation && error.exc_info.extra) {
+      this.handleValidationError({ ...error, extra: error.exc_info.extra as Job['extra'] }, formGroup, fieldsMap);
+      return;
+    }
+
+    // Fallback to old error handling
+    this.dialog.error(this.parseErrorOrJob(error));
+  }
+
+  private handleValidationError(
+    error: WebsocketError | Job,
+    formGroup: UntypedFormGroup,
+    fieldsMap: Record<string, string>,
+  ): void {
+    for (const extraItem of (error as WebsocketError).extra) {
+      const field = extraItem[0].split('.')[1];
+      const errorMessage = extraItem[1];
+
+      const control = this.getFormField(formGroup, field, fieldsMap);
+      if (!control) {
+        console.error(`Could not find control ${field}.`);
+        // Fallback to default modal error message.
+        this.dialog.error(this.parseErrorOrJob(error));
+        return;
+      }
+
+      control.setErrors({
+        manualValidateError: true,
+        manualValidateErrorMsg: errorMessage,
+        ixManualValidateError: { message: errorMessage },
+      });
+      control.markAsTouched();
+    }
+  }
+
+  reportWsError(error: WebsocketError): Observable<boolean> {
+    return this.dialog.error(this.parseWsError(error));
+  }
+
+  parseWsError(error: WebsocketError): ErrorReport | ErrorReport[] {
+    return this.parseErrorOrJob(error);
+  }
+
+  reportJobError(failedJob: Job): Observable<boolean> {
```

Comment:
`parseWsError` is the same as `parseErrorOrJob`, so why not just rename the latter?
Also since both `reportWsError` and `reportJobError` essentially do the same thing, we don't really need the distinction.